src/main/java/com/teragrep/jos_01/procfs/Process.java

```
@@ -0,0 +1,185 @@
+import org.slf4j.LoggerFactory;
+
+public class Process {
+
+    private final long processId;
+    private final File procDirectory;
+    private final Logger LOGGER = LoggerFactory.getLogger(Process.class);
+
+    public Process(String processId) {
+        this(Integer.parseInt(processId));
+    }
+
+    public Process(long processId) {
+        this(processId, "/proc");
+    }
+
+    public Process(long processId, String procDirectoryPath) {
+        this.processId = processId;
+        this.procDirectory = new File(procDirectoryPath, Long.toString(processId));
+    }
+
+    // Creates a Status object based on the chosen file name in /proc for this process. Overloaded to accept different kinds of Status objects
+    private ArrayList<String> reportStatistics(String procFileName) throws IOException {
+        ArrayList<String> rows = readProcFile(procFileName);
+        return rows;
+    }
+
+    private ArrayList<String> readProcFile(String procFileName) throws IOException {
+        RowFile procFile = new RowFile(procDirectory, procFileName);
+        ArrayList<String> rows = procFile.readFile();
+        return rows;
+    }
+
+    public ProcessStat stat() throws IOException {
+        ArrayList<String> rows = readProcFile("stat");
+        return new ProcessStat(rows);
+    }
+
+    public Statm statm() throws IOException {
+        ArrayList<String> rows = readProcFile("statm");
+        return new Statm(rows);
+    }
+
+    public ArrayList<String> availableProcFiles() {
+        ArrayList<String> fileNames = procFileNames(new ArrayList<String>(), procDirectory);
+        return fileNames;
+    }
+
+    private ArrayList<String> procFileNames(ArrayList<String> nameList, File file) {
+        if (!file.isDirectory()) {
+            nameList.add(file.getPath().replace(procDirectory.getPath(), ""));
+        }
+        else {
+            File[] subdirectories = file.listFiles();
+            if (subdirectories == null) {
+                LOGGER
```

throw IOException instead of logging